homeworks/week5/trang_vu

```diff
@@ -0,0 +1,49 @@
+import java.util.HashMap;
+
+class trang_vu5 {
+  public static void main(String[] args) {
+    int[] boss = { 1, 3, 3, -1, -1, 4, 2, 8, 3 };
+    System.out.println(lowestCommonBoss(boss, 1, 1)); // 1
+    System.out.println(lowestCommonBoss(boss, 2, 0)); // 3
+    System.out.println(lowestCommonBoss(boss, 6, 0)); // 3
+    System.out.println(lowestCommonBoss(boss, 7, 1)); // 3
+    System.out.println(lowestCommonBoss(boss, 4, 5)); // 4
+    System.out.println(lowestCommonBoss(boss, 6, 5)); // -1
+    System.out.println(lowestCommonBoss(boss, 4, 2)); // -1
+    System.out.println(lowestCommonBoss(boss, 4, 3)); // -1
+    System.out.println(lowestCommonBoss(boss, 4, -1)); // -1
+  }
+
+
+  // Time complexity : O(logn), travel up to 2 times height  = log2(n) x2
+  // Space complexity: 0(1), do not require additional space
```

Comment:
This is not correct. The method uses path1 with O(logN) space in average.

Please see sample solution for an optimal solution.